### PR TITLE
Js/nc groebner basis documentation

### PR DESCRIPTION
**Pull request by julien-schanz**
Expanded the documentation for the non-commutative groebner bases for free associative algebras.

### CODE REVIEW

File: docs/src/free_associative_algebra.md [hunk 1]
```diff
@@ -229,6 +229,19 @@ generators of that ideal.
...
+    normal_form(f::FreeAssAlgElem{T}, g::Vector{FreeAssAlgElem{T}}, aut::AhoCorasickAutomaton) where T
+
+    interreduce!(g::Vector{FreeAssAlgElem{T}}) where T
+```
+
+The implementation uses a non-commutative version of the Buchberger algorithm as described in
+> Xingqiang Xiu,
+> Non-commutative Gröbner Bases and Applications,
+> PhD thesis, 2012.
+
```
Comment (joschmitt) [inline docs/src/free_associative_algebra.md:244]: You can add this as a proper bibtex reference in `doc/oscar_references.bib` and then cite it here with `[LABEL](@cite)`.

Comment (joschmitt) [reply docs/src/free_associative_algebra.md:244]: Sorry, wrong repository.

**Comment by fingolfin** [comment]
Do you still have plans to improve the code? E.g. the reduction code still is substantially slower than it could be as it uses `_sub_rest` (instead of geobuckets or the even simpler approach of using a hash table plus a heap (as e.g. described in section 4.1 of <https://arxiv.org/abs/1206.6940>).

**Review by fingolfin — APPROVED** [review]

File: docs/src/free_associative_algebra.md [hunk 2]
```diff
@@ -229,6 +229,19 @@ generators of that ideal.
 Since such a Groebner basis is not necessarily finite, one can additionally pass a `reduction_bound`
 to the function, to only compute a partial Groebner basis.
 
+```@docs
+    groebner_basis(g::Vector{FreeAssAlgElem{T}}, reduction_bound::Int = typemax(Int), remove_redundancies::Bool = false) where T <: FieldElement
+
+    normal_form(f::FreeAssAlgElem{T}, g::Vector{FreeAssAlgElem{T}}, aut::AhoCorasickAutomaton) where T
+
+    interreduce!(g::Vector{FreeAssAlgElem{T}}) where T
```
Comment (lgoettgens) [inline docs/src/free_associative_algebra.md:237]:
```suggestion
groebner_basis(g::Vector{FreeAssAlgElem{T}}, reduction_bound::Int = typemax(Int), remove_redundancies::Bool = false) where T <: FieldElement

normal_form(f::FreeAssAlgElem{T}, g::Vector{FreeAssAlgElem{T}}, aut::AhoCorasickAutomaton) where T

interreduce!(g::Vector{FreeAssAlgElem{T}}) where T
```

**Comment by julien-schanz** [comment]
@fingolfin Currently, I do not have any capacity to continue working on the code, so I am not planning to improve anything else at the moment.

**Comment by fingolfin** [comment]
Thanks for the clarification @julien-schanz .

I'll happily merge this, but the real issue with making these docs visible to OSCAR users is to make sure it is linked to in the OSCAR manual, see [this OSCAR issue](https://github.com/oscar-system/Oscar.jl/issues/3924) which I just opened.

**Pull request merged**